packages/react/src/Hint/Hint.tsx

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright Gemeente Amsterdam
+ */
+
+import clsx from 'clsx'
+import { forwardRef } from 'react'
+import type { ForwardedRef, HTMLAttributes, PropsWithChildren } from 'react'
+
+export type HintProps = PropsWithChildren<HTMLAttributes<HTMLElement>> & {
+  /** Show a custom hint text. */
+  hint?: string
+  /** Appends the text '(niet verplicht)' to the label or legend if no hint is provided. Use when the associated inputs are optional. */
+  optional?: boolean
+}
+
+const getHintText = ({ hint, optional }: HintProps) => {
```

Comment:
Not the most important thing in the world, but: I think this allows all global HTML attributes to be passed to this function. I think it's cleaner to separate the `hint` and `optional` type from `PropsWithChildren<HTMLAttributes<HTMLElement>>`, and only use the first one here.

Reply:
Aangepast